test/Microsoft.AspNet.Mvc.Core.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -858,6 +864,352 @@ public void ListBoxFor_WithUnreleatedExpression_GeneratesExpectedValue()
             Assert.Equal(savedSelected, selectList.Select(item => item.Selected));
         }
 
+        [Fact]
+        public void GetSelectListTEnum_ThrowsWithFlagsEnum()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => htmlHelper.GetSelectList<EnumWithFlags>(),
+                "TEnum",
+                $"The type '{ typeof(EnumWithFlags).FullName }' is not supported.");
+        }
+
+        [Fact]
+        public void GetSelectListTEnum_ThrowsWithNonEnum()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act & Assert
+            ExceptionAssert.ThrowsArgument(
+                () => htmlHelper.GetSelectList<StructWithFields>(),
+                "TEnum",
+                $"The type '{ typeof(StructWithFields).FullName }' is not supported.");
+        }
+
+        [Fact]
+        public void GetSelectListTEnum_WrapsGetSelectListModelMetadata()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var metadata = metadataProvider.GetMetadataForType(typeof(EnumWithFields));
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act
+            var result = htmlHelper.GetSelectList<EnumWithFields>();
+
+            // Assert
+            Assert.Equal(metadata.ModelType, htmlHelper.Metadata.ModelType);
```

Comment:
wait, how does this happen? I'm confused about how `htmlHelper.Metadata` get its type